internal/controller/nnf_node_block_storage_controller.go

```diff
@@ -644,7 +684,275 @@ func (r *NnfNodeBlockStorageReconciler) getStorageGroup(ss nnf.StorageServiceApi
 }
 
 func (r *NnfNodeBlockStorageReconciler) deleteStorageGroup(ss nnf.StorageServiceApi, id string) error {
-	return ss.StorageServiceIdStorageGroupIdDelete(ss.Id(), id)
+	err := ss.StorageServiceIdStorageGroupIdDelete(ss.Id(), id)
+	if err != nil {
+		ecErr, ok := err.(*ec.ControllerError)
+		// If the error is from a 404 error, treat as success (idempotent deletion)
+		if ok && ecErr.StatusCode() == http.StatusNotFound {
+			return nil
+		}
+	}
+	return err
+}
+
+// getComputeNodesWithAccess returns a list of compute node names that have access to this NnfNodeBlockStorage
+func (r *NnfNodeBlockStorageReconciler) getComputeNodesWithAccess(blockStorage *nnfv1alpha9.NnfNodeBlockStorage) []string {
+	computeNodes := make(map[string]bool)
+
+	for _, allocation := range blockStorage.Status.Allocations {
+		if allocation.Accesses != nil {
+			for computeNode := range allocation.Accesses {
+				computeNodes[computeNode] = true
+			}
+		}
+	}
+
+	result := make([]string, 0, len(computeNodes))
+	for computeNode := range computeNodes {
+		result = append(result, computeNode)
+	}
+	return result
+}
+
+// scanFenceRequests scans the fence request directory for requests targeting any of the given compute nodes
+func (r *NnfNodeBlockStorageReconciler) scanFenceRequests(computeNodes []string, log logr.Logger) ([]*FenceRequest, error) {
+	// Check if fence request directory exists
+	if _, err := os.Stat(fence.RequestDir); os.IsNotExist(err) {
+		return nil, nil
+	}
+
+	// Create a map for fast lookup
+	computeNodeMap := make(map[string]bool)
+	for _, node := range computeNodes {
+		computeNodeMap[node] = true
+	}
+
+	// Read all files in the request directory
+	entries, err := os.ReadDir(fence.RequestDir)
+	if err != nil {
+		return nil, err
+	}
+
+	var fenceRequests []*FenceRequest
+
+	// Check each request file to see if it's for this compute node
+	for _, entry := range entries {
+		if entry.IsDir() || filepath.Ext(entry.Name()) != ".json" {
+			continue
+		}
+
+		requestFile := filepath.Join(fence.RequestDir, entry.Name())
+		data, err := os.ReadFile(requestFile)
+		if err != nil {
+			log.Error(err, "Failed to read fence request file", "file", requestFile)
```

Comment:
I'm assuming that you're not returning errors unless absolutely necessary. Maybe a comment about why?